### ISSUE TITLE
Move some exercises up in episodes

### ISSUE THREAD

**Issue by dafnevk** [swcarpentry/python-novice-gapminder#564]
I have a rather generic remark about this lesson that applies to multiple episodes.
In all episodes, all exercises are grouped at the end of the episode. When teaching, I find it useful to insert a small exercise immediately after I introduce a concepts. Therefore, I think some of the exercises can be moved up in the episodes, immediately following an explanation. The more advanced challenges that combine the different concepts in the episode (and that might be suited for group / breakout room exercises) can be put at the end of the episode.

Happy to hear others thoughts on this, I could try to make a PR to rearrange one of the early episodes in this manner.

**Comment by jdinklo** [comment]
Yes, I too find it very useful for students to be able to practice what you have just demonstrated. I am looking at teaching the 'The Unix Shell' which does have a small exercise after the concept is explained. Perhaps it depends on the course, as to the order in which the exercises are presented?

**Comment by ferproanocuenca** [comment]
I have sat in some workshops and find that most instructors use the exercises immediately after introducing a new concept. Other times, the exercises are used later once more information is available to push attendees to interact more or be more involved. So it depends on the teaching style, the lesson's topic, and the time to fit in the schedule.

Anyhow, I agree it depends on the course and possibly can be moved up in some cases.
